tests/src/tests.rs

```diff
@@ -656,9 +692,54 @@ struct TestPartMetadata {
 
 #[derive(Debug, Clone, Eq, PartialEq, Hash)]
 struct Annotation {
-    range: Option<Range<usize>>,
     message: EcoString,
     kind: AnnotationKind,
+    target: AnnotationTarget,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+enum AnnotationTarget {
+    Local(Option<Range<usize>>),
```

Comment:
@\reviewer it might be possible to make the range non-optional, that would make sense, right? There's the `Detached` case for this now

Reply:
https://github.com/typst/typst/pull/1872/commits/639ab44964b8838506d542f2ab19513d5bfe1b5b